transforms/universal/pdf2parquet/python/src/pdf2parquet_transform.py

```
@@ -0,0 +1,269 @@
+                                if kind is None or kind.mime != "application/pdf":
+                                    logger.info(
+                                        f"File {archive_doc_filename=} is not detected as PDF but {kind=}. Skipping."
+                                    )
+                                    skipped_doc_id.append(archive_doc_filename)
+                                    continue
+
+                                ext = kind.extension
+
+                                file_data = self._convert_pdf2parquet(doc_filename=archive_doc_filename, ext=ext, content_bytes=content_bytes)
+                                file_data["source_filename"] = TransformUtils.get_file_basename(file_name)
+
+                                data.append(file_data)
+                                success_doc_id.append(archive_doc_filename)
+                                number_of_rows += 1
+
+                            except Exception as e:
+                                failed_doc_id.append(archive_doc_filename)
+                                logger.warning(
+                                    f"Exception {str(e)} processing file {archive_doc_filename}, skipping"
+                                )
+            
+            else:
+                logger.warning(
+                    f"File {file_name=} is not detected as PDF nor as ZIP but {kind=}. Skipping."
+                )
+
+            table = pa.Table.from_pylist(data)
+            metadata = {
+                "nrows": len(table),
+                "nsuccess": len(success_doc_id),
+                "nfail": len(failed_doc_id),
+                "nskip": len(skipped_doc_id),
+            }
+            return [
+                (TransformUtils.convert_arrow_to_binary(table=table), ".parquet")
+            ], metadata
+        except Exception as e:
+            logger.error(f"Fatal error with file {file_name=}. No results produced.")
```

You should throw an exception instead of eating it.  The runtime will log it and may keep a count.  The result will be the same, that is, that there is no file written for this call to transform().

---

fixed